File: Tests/Testing/AngularTest/TestApi3/src/services/misc.service.ts
import { Injectable } from '@nestjs/common';
import { PersonDTO } from './dto/People/person.dto';
import { PersonUpdateDTO } from './dto/People/person-update.dto';
import { DateDTO } from './dto/Misc/date.dto';
import { IntentIgnoreBody } from './../intent/intent.decorators';
import { randomUUID } from 'crypto';

@Injectable()
export class MiscService {

  //@IntentCanAdd()
  constructor() { }

  @IntentIgnoreBody()
  async getWithQueryParam(idParam: string): Promise<PersonDTO> {
    return {
      id: randomUUID(),
      name: 'Test ' + randomUUID()
    };
  }

  @IntentIgnoreBody()
  async getWithRouteParam(routeId: string): Promise<PersonDTO> {
    return {
      id: randomUUID(),
      name: 'Test ' + randomUUID()
    };
  }

  @IntentIgnoreBody()
  async postWithHeaderParam(param: string): Promise<void> {
    
  }

  @IntentIgnoreBody()
  async postWithBodyParam(param: PersonUpdateDTO): Promise<void> {
    
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultInt(): Promise<number> {
    return 12;
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultWrapInt(): Promise<number> {
    return 13;
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultBool(): Promise<boolean> {
    return true;
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultWrapBool(): Promise<boolean> {
    return false;
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultStr(): Promise<string> {
    return "Primitive string value";
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultWrapStr(): Promise<string> {
    return "Wrapped Primitive string value";
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultDate(): Promise<Date> {
    return new Date();
  }

  @IntentIgnoreBody()
  async getWithPrimitiveResultWrapDate(): Promise<Date> {
    return new Date();
  }

  @IntentIgnoreBody()
  async postDateParam(date: Date, datetime: Date): Promise<void> {
    
  }

  @IntentIgnoreBody()
  async postDateParamDto(dto: DateDTO): Promise<void> {
    
  }
}

File: Tests/Testing/AngularTest/TestApi2/src/main/java/com/AngularTest/TestApi2/application/services/MiscService.java
package com.AngularTest.TestApi2.application.services;

import com.AngularTest.TestApi2.application.models.Misc.DateDTO;
import com.AngularTest.TestApi2.application.models.People.PersonDTO;
import com.AngularTest.TestApi2.application.models.People.PersonUpdateDTO;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;


public interface MiscService {
    PersonDTO GetWithQueryParam(UUID idParam);

    PersonDTO GetWithRouteParam(UUID routeId);

    void PostWithHeaderParam(String param);

    void PostWithBodyParam(PersonUpdateDTO param);

    int GetWithPrimitiveResultInt();

    int GetWithPrimitiveResultWrapInt();

    boolean GetWithPrimitiveResultBool();

    boolean GetWithPrimitiveResultWrapBool();

    String GetWithPrimitiveResultStr();

    String GetWithPrimitiveResultWrapStr();

    LocalDateTime GetWithPrimitiveResultDate();

    LocalDateTime GetWithPrimitiveResultWrapDate();

    void PostDateParam(LocalDate date, LocalDateTime datetime);

    void PostDateParamDto(DateDTO dto);

}